src/hierarchies/lapnig_hierarchy.h

```diff
@@ -0,0 +1,125 @@
+  bayesmix::HierarchyId get_id() const override {
+    return bayesmix::HierarchyId::LapNIG;
+  }
+
+  //! Read and set state values from a given Protobuf message
+  void set_state_from_proto(const google::protobuf::Message &state_) override;
+
+  //! Read and set hyperparameter values from a given Protobuf message
+  void set_hypers_from_proto(
+      const google::protobuf::Message &hypers_) override;
+
+  void save_posterior_hypers() { throw std::runtime_error("Not implemented"); }
+
+  //! Returns whether the hierarchy models multivariate data or not
+  bool is_multivariate() const override { return false; }
+
+  //! Writes current state to a Protobuf message and return a shared_ptr
+  //! New hierarchies have to first modify the field 'oneof val' in the
+  //! AlgoritmState::ClusterState message by adding the appropriate type
+  std::shared_ptr<bayesmix::AlgorithmState::ClusterState> get_state_proto()
+      const override;
+
+  //! Writes current value of hyperparameters to a Protobuf message and
+  //! return a shared_ptr.
+  //! New hierarchies have to first modify the field 'oneof val' in the
+  //! AlgoritmState::HierarchyHypers message by adding the appropriate type
+  std::shared_ptr<bayesmix::AlgorithmState::HierarchyHypers> get_hypers_proto()
+      const override;
+
+  //! Resets summary statistics for this cluster
+  void clear_summary_statistics() override;
+
+  //! Updates hyperparameter values given a vector of cluster states
+  void update_hypers(const std::vector<bayesmix::AlgorithmState::ClusterState>
+                         &states) override;
+
+  //! Updates state values using the given (prior or posterior) hyperparameters
+  LapNIG::State draw(const LapNIG::Hyperparams &params);
+
+  //! Generates new state values from the centering posterior distribution
+  //! @param update_params  Save posterior hypers after the computation?
+  void sample_full_cond(bool update_params = false) override;
+
+ protected:
+  //! Set of values of data points belonging to this cluster
+  std::list<Eigen::RowVectorXd> cluster_data_values;
+
+  //! Sum of absolute differences for current params
+  double sum_abs_diff_curr = 0;
+
+  //! Sum of absolute differences for proposal params
+  double sum_abs_diff_prop = 0;
+
+  //! Samples from the proposal distribution using Random Walk
+  //! Metropolis-Hastings
+  Eigen::VectorXd propose_rwmh(const Eigen::VectorXd &curr_vals);
+
+  //! Evaluates the prior given the mean (unconstrained_parameters(0))
+  //! and log of the scale (unconstrained_parameters(1))
+  double eval_prior_lpdf_unconstrained(
+      Eigen::VectorXd unconstrained_parameters);
+
+  //! Evaluates the (sum of the) log likelihood for all the observations in the
+  //! cluster given the mean (unconstrained_parameters(0))
+  //! and log of the scale (unconstrained_parameters(1))
+  double eval_like_lpdf_unconstrained(Eigen::VectorXd unconstrained_parameters,
+                                      bool is_current);
```

Comment:
What does `is_current` mean? You should briefly explain it in the documentation comment

Reply:
We used "is_current" to identify if the evaluation of the likelihood is on the current or on the proposed parameters, in order to avoid repeating calculations of the sum of the absolute differences. We added a comment above explaining it. Thank you :)